### PR TITLE
Webhooks guide 1: Discussion bot powered by BLOOM

### PR DESCRIPTION
**Pull request by julien-c**
(i'll move the assets before merging)

### CODE REVIEW

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Webhook guide: build a Discussion bot based on BLOOM
+
+Here's a short guide on how to use HF Webhooks to build a bot that replies to Discussion comments on the Hub with a response generated by BLOOM (using the Inference API).
+
+## Create your webhook in your user profile
+
+First, create your webhook by going to https://huggingface.co/settings/webhooks
```
Comment (Michellehbn) [inline docs/hub/webhooks-guide-discussion-bot.md:7]:
```suggestion
First, create your webhook by going to https://huggingface.co/settings/webhooks.
```

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+- Input a few target repositories that your webhook will run on.
+- You can put a dummy Webhook URL for now, but defining your webhook will let you take a look at the events that are going to be sent to it (and you can replay them, which will be useful for debugging).
+- Input a secret as it will be more secure.
+- subscribe to Community (PR & discussions) events as we are building a Discussion bot
```
Comment (Michellehbn) [inline docs/hub/webhooks-guide-discussion-bot.md:12]:
```suggestion
- Subscribe to Community (PR & discussions) events, as we are building a Discussion bot.
```

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		headers: {
+			Authorization: `Bearer ${process.env.HF_TOKEN}`,
+			"Content-Type": "application/json",
+		},
+		body: JSON.stringify({ comment: continuationText }),
+	});
+
+	const apiOutput = await commentApiResponse.json();
+```
+
```
Comment (SBrandeis) [inline docs/hub/webhooks-guide-discussion-bot.md:109]: Maybe worth mentioning the user needs to update the webhook to fetch the space's URL

**Review by coyotte508 — APPROVED** [review]
Very nice!

File: docs/hub/webhooks-guide-discussion-bot.md (hunk 3 above)
Comment (SBrandeis) [reply docs/hub/webhooks-guide-discussion-bot.md:109]: Is the "raw" space URL documented somewhere? `***.hf.space`

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+- Input a secret as it will be more secure.
+- subscribe to Community (PR & discussions) events as we are building a Discussion bot
+
+Your webhook will look like this:
+
+![webhook-creation](../assets/webhook-creation.png)
+
+## Create a new "Bot" user profile
+
+In this guide, we create a separate user account to host a Space and to post comments.
```
Comment (Michellehbn) [inline docs/hub/webhooks-guide-discussion-bot.md:20]:
```suggestion
In this guide, we create a separate user account to host a Space and to post comments:
```

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+![discussion-bot-profile](../assets/discussion-bot-profile.png)
+
+<Tip>
+	When creating a bot that will interact with other users on the Hub, we ask that you clearly label the account as a Bot (see profile screenshot).
+</Tip>
+
+## Create a Space that will react to your webhook
```
Comment (SBrandeis) [inline docs/hub/webhooks-guide-discussion-bot.md:28]: Could be good to link to the Docker space doc too

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+![webhook-creation](../assets/webhook-creation.png)
+
+## Create a new "Bot" user profile
+
+In this guide, we create a separate user account to host a Space and to post comments.
+
+![discussion-bot-profile](../assets/discussion-bot-profile.png)
+
+<Tip>
+	When creating a bot that will interact with other users on the Hub, we ask that you clearly label the account as a Bot (see profile screenshot).
```
Comment (Michellehbn) [inline docs/hub/webhooks-guide-discussion-bot.md:25]:
```suggestion
	When creating a bot that will interact with other users on the Hub, we ask that you clearly label the account as a "Bot" (see profile screenshot).
```

File: docs/hub/webhooks-guide-discussion-bot.md (hunk 5 above)
Comment (SBrandeis) [reply docs/hub/webhooks-guide-discussion-bot.md:28]: OK I see there's a link in the Dockerfile 👍

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+</Tip>
+
+## Create a Space that will react to your webhook
+
+The third step is to actually listen to the webhook events.
```
Comment (Michellehbn) [inline docs/hub/webhooks-guide-discussion-bot.md:30]:
```suggestion
The third step is to listen to the webhook events.
```

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+An easy way is just to use a Space for this. We use the user account we just created, but you could do it from your main user account if you wanted to.
```
Comment (Michellehbn) [inline docs/hub/webhooks-guide-discussion-bot.md:32]:
```suggestion
An easy way is to use a Space for this. We will use the user account we just created, but you could do it from your main user account if you wanted to.
```

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+The Space's code is here: https://huggingface.co/spaces/discussion-bot/webhook/tree/main
```
Comment (coyotte508) [inline docs/hub/webhooks-guide-discussion-bot.md:34]: Maybe mention that you can duplicate it if you want?

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+	if (response.ok) {
+		const output = await response.json();
+		const continuationText = output[0].generated_text.replace(
+			PROMPT + req.body.comment.content,
+			""
+		);
+		...
+```
+
+This is the coolest part: we call the Inference API for the BLOOM model, prompting it with `PROMPT`, and we get the continuation text i.e. the part generated by the model.
```
Comment (Michellehbn) [inline docs/hub/webhooks-guide-discussion-bot.md:91]:
```suggestion
This is the coolest part: we call the Inference API for the BLOOM model, prompting it with `PROMPT`, and we get the continuation text, i.e. the part generated by the model.
```

File: docs/hub/webhooks.md [hunk 11]
```diff
@@ -193,3 +193,13 @@ Go in the activity tab for your webhook, there you will see the list of recent e
 You will see the HTTP status code and the payload of past events. You can replay those events by clicking on the `replay` button!
 
 When changing the target URL or secret of a webhook, replaying an event will send the payload to the updated URL.
+
+## FAQ
+
+##### Can I define webhooks on my organization vs. on my user account?
```
Comment (Michellehbn) [inline docs/hub/webhooks.md:199]:
```suggestion
##### Can I define webhooks on my organization vs my user account?
```

File: docs/hub/webhooks.md [hunk 12]
```diff
@@ -193,3 +193,13 @@ Go in the activity tab for your webhook, there you will see the list of recent e
...
+
+No, this is not currently supported.
+
+##### How can I subscribe to events on all repos (or across a whole repo type, like on all models)
```
Comment (Michellehbn) [inline docs/hub/webhooks.md:203]:
```suggestion
##### How can I subscribe to events on all repos (or across a whole repo type, like on all models)?
```

**Review by Michellehbn — APPROVED** [review]
LGTM! 🔥

File: docs/hub/webhooks-guide-discussion-bot.md (hunk 3 above)
Comment (julien-c) [reply docs/hub/webhooks-guide-discussion-bot.md:109]: yes you're completely right. Forgot this.

File: docs/hub/webhooks-guide-discussion-bot.md (hunk 5 above)
Comment (julien-c) [reply docs/hub/webhooks-guide-discussion-bot.md:28]: but i agree we should add it here too

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
+# Webhook guide: build a Discussion bot based on BLOOM
+
+Here's a short guide on how to use HF Webhooks to build a bot that replies to Discussion comments on the Hub with a response generated by BLOOM (using the Inference API).
+
+## Create your webhook in your user profile
+
+First, create your webhook by going to https://huggingface.co/settings/webhooks
+
+- Input a few target repositories that your webhook will run on.
```
Comment (merveenoyan) [inline docs/hub/webhooks-guide-discussion-bot.md:9]: Pardon me if I want to dumb it down but can you perhaps clarify what it will do in the repository exactly?

File: docs/hub/webhooks-guide-discussion-bot.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+The Space's code is here: https://huggingface.co/spaces/discussion-bot/webhook/tree/main
+
+We used NodeJS and Typescript to implement it, but any language or framework would work equally well.
+
+**The main `server.ts` file is [here](https://huggingface.co/spaces/discussion-bot/webhook/blob/main/server.ts)**
+
+Let's walk through what happens in this file:
+
+```ts
```
Comment (merveenoyan) [inline docs/hub/webhooks-guide-discussion-bot.md:42]: Can't we do it in python directly given our audience?

File: docs/hub/webhooks.md (hunk 11 above)
Comment (merveenoyan) [inline docs/hub/webhooks.md:199]: I couldn't understand the question here, can you specify? e.g. Can I define webhooks on my organization?
(is it like you define webhooks defined from your org, affecting your user repositories?)

**Review by merveenoyan — COMMENTED** [review]
Left nits on ambiguity 🙂

File: docs/hub/webhooks-guide-discussion-bot.md (hunk 13 above)
Comment (julien-c) [reply docs/hub/webhooks-guide-discussion-bot.md:9]: will rephrase

File: docs/hub/webhooks-guide-discussion-bot.md (hunk 14 above)
Comment (merveenoyan) [reply docs/hub/webhooks-guide-discussion-bot.md:42]: resolving this since @SBrandeis made one

File: docs/hub/webhooks.md (hunk 11 above)
Comment (merveenoyan) [reply docs/hub/webhooks.md:199]: `Can I define webhooks on my organization's settings like I can with my user's account?` does it sound good 😅

Comment (merveenoyan) [inline docs/hub/webhooks.md:199]:
```suggestion
##### Can I define webhooks on my organization's settings like I can with my user's account?
```

File: docs/hub/webhooks-guide-discussion-bot.md (hunk 3 above)
Comment (julien-c) [reply docs/hub/webhooks-guide-discussion-bot.md:109]: added in 0497401
